polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -1570,13 +1572,11 @@ ValueCategory MLIRScanner::VisitCastExpr(CastExpr *E) {
       return ValueCategory(Nval, /*isReference*/ false);
     }
 
-#ifdef DEBUG
```

Comment:
What's the motivation to remove these `#ifdef DEBUG`? Should we guard them by `LLVM_DEBUG`?

Reply:
In other points of the programs, constructs like these (printing involved values that will miss a following `assert`) are not guarded, so I just followed that approach. IMO, we could just remove all of these completely, i.e., no printing before asserts.

Reply:
oic, didn't notice they `assert` after print, I am fine with those.
How about the ones at line 1721 and 1781, should we put them in `LLVM_DEBUG`?

Reply:
I am fine removing the guards for the prints that assert after, as we shouldn't see them when compile successfully. My concern is on the prints that show up even if we compile successfully. e.g., the one at line 1721:
```
#ifdef DEBUG
    if (!Scalar.getType().isa<mlir::IntegerType>()) {
      E->dump();
      llvm::errs() << " Scalar: " << Scalar << "\n";
    }
#endif

    auto PrevTy = Scalar.getType().cast<mlir::IntegerType>();
    bool SignedType = true;
    if (const auto *Bit =
            dyn_cast<clang::BuiltinType>(&*E->getSubExpr()->getType())) {
...
```
Am I missing something?

Reply:
Sorry, I was not clear. We're checking type correctness in the `if` and then `cast`ing. `cast` will throw an error if the type is incorrect.

Reply:
True, thanks for clarifying.